packages/telemetry/src/slog-to-otel.js

```diff
@@ -104,13 +110,124 @@ export const makeSlogToOtelKit = (tracer, overrideAttrs = {}) => {
   const dbTransactionManager = makeKVDatabaseTransactionManager(db);
   const kernelPromiseToSendingCause = makeKVStringStore('kernelPromise', db);
 
-  const extractMethod = methargs => JSON.parse(methargs.body)[0];
+  /** @type {WeakMap<Promise | object, {kref: string, iface: string | undefined}>} */
+  const presenceMap = new WeakMap();
+  /** @type {Map<string, Promise | object>} */
+  const valCache = new Map();
+
+  const slotToVal = (kref, iface) => {
+    let val = valCache.get(kref);
+    if (val) {
+      return val;
+    }
+
+    if (iface) {
+      if (!iface.startsWith('Alleged: ') && iface !== 'Remotable') {
+        iface = `Alleged: ${iface}`;
+      }
+    }
+
+    kref[0] === 'k' || assert.fail(`Unexpected non-kernel ref ${kref}`);
+    val = harden(
+      kref.startsWith('kp')
+        ? new Promise(sink)
+        : Remotable(iface || 'Remotable'),
+    );
+    valCache.set(kref, val);
+    presenceMap.set(val, { kref, iface });
+
+    return val;
+  };
+
+  const valToSlot = val => {
+    const { kref } = presenceMap.get(val) || {};
+    kref || assert.fail(`Unknown val ${val}`);
+    return kref;
+  };
+
+  const { unserialize: rawUnserialize } = makeMarshal(valToSlot, slotToVal, {
```

Comment:
`valToSlot` is not needed if the only thing you're creating is `unserialize`.

```suggestion
  const { unserialize: rawUnserialize } = makeMarshal(undefined, slotToVal, {
```